lib/_http_client.js

```diff
@@ -35,6 +34,13 @@ function ClientRequest(options, cb) {
   } else if ((agent === null || agent === undefined) &&
              typeof options.createConnection !== 'function') {
     agent = defaultAgent;
+  } else if (agent !== null && agent !== undefined &&
+            !(agent instanceof Agent.Agent)) {
+    console.log(agent, typeof agent);
+    throw new Error(
+      'Agent option must be an instance of http.Agent, undefined or false. '
+      + 'Received type: ' + typeof agent + ' instead'
```

Comment:
Tiny nit: can you indent by four spaces here?  It's allowed to use string templates (backticks) if you think that's easier to read.